### PR TITLE
fix: add subscribe to notification to settings

### PR DESCRIPTION
**Pull request by schalkneethling**
Add the ability to enable in-app notifications from settings.

## Success

Screenshot 2021-05-17 at 08 14 41

### Failed

Screenshot 2021-05-17 at 08 14 05

### CODE REVIEW

**Comment by otterbotter** [comment]
I opened the deploy preview and went to settings but I don't see any in-app notifications setting.
Does it only show if you do not have them enabled?

no-in-app-notification-setting

File: src/js/components/account/user-settings.js [hunk 1]
```diff
@@ -24,6 +26,14 @@ export class UserSettings {
           "Change Password"
         );
         $container.append($changePasswordAnchor);
+
+        if (pushpad) {
+          pushpad("status", (isSubscribed) => {
+            if (isSubscribed) {
```
Comment (otterbotter) [inline src/js/components/account/user-settings.js:32]: shouldn't this conditional be `if (!isSubscribed)` ?

**Review by otterbotter — COMMENTED** [review]
Reviewed docs (https://app.gitbook.com/@openup/s/cape-agulhas-app/user-guide/reset-notification-preferences) - they look good! Great job!

**Comment by otterbotter** [comment]
I get a TypeError when opening the user settings page

pushpad-is-not-defined

**Comment by schalkneethling** [comment]
> Shouldn't the setting always be visible? Or at least visible if you do not have notifications enabled?

It only shows if you do not already have a subscription.

File: src/js/components/account/user-settings.js (hunk 1 above)
Comment (schalkneethling) [reply src/js/components/account/user-settings.js:32]: Doh! I swapped it when testing. 😸

**Comment by otterbotter** [comment]
It might be confusing for users to sometimes see the setting and sometimes not- would it be difficult to always show the notification setting and allow them to both `subscribe` and `unsubscribe`? 

If they visited the settings page to enable their notifications, they might revisit later to disable them but not see the setting anymore.

**Comment by schalkneethling** [comment]
> It might be confusing for users to sometimes see the setting and sometimes not- would it be difficult to always show the notification setting and allow them to both `subscribe` and `unsubscribe`?
> 
> If they visited the settings page to enable their notifications, they might revisit later to disable them but not see the setting anymore.

I was wondering the same thing but did not want to scope creep :) I reckon it is worth adding.

**Comment by schalkneethling** [comment]
> I get a TypeError when opening the user settings page (`user-settings.js:30 Uncaught ReferenceError: pushpad is not defined`)

I am not getting this. Are you running `yarn dev` or `yarn start`? Does not really matter though. I mean, that is the line where the conditional is run. If it is undefined, the condition should evaluate as false and skip the contents of the condition.

**Comment by otterbotter** [comment]
> > I get a TypeError when opening the user settings page (`user-settings.js:30 Uncaught ReferenceError: pushpad is not defined`)
> 
> I am not getting this. Are you running `yarn dev` or `yarn start`? Does not really matter though. I mean, that is the line where the conditional is run. If it is undefined, the condition should evaluate as false and skip the contents of the condition.

Neither, it's from the deploy preview. It's because there's no pushpad environment variables configured. Not ideal to have it raise an exception but you are right in that it still evals as falsy and continues so it's fine 👍🏼

**Comment by schalkneethling** [comment]
> It's because there's no pushpad environment variables configured.

Understood, but I commented my setting out of my local `env` and so, there should be no pushpad id either locally no? Do we _always_ have a pushpad id set locally?

**Comment by otterbotter** [comment]
> > It's because there's no pushpad environment variables configured.
> 
> Understood, but I commented my setting out of my local `env` and so, there should be no pushpad id either locally no? Do we _always_ have a pushpad id set locally?

No if you comment it out it shouldn't have an ID, but you would need to restart the `dev` script.

If you have properly disabled pushpad it will print a warning to the console: `Pushpad project ID not set; not initialising` then you know there's definitely no project ID set. But it's really fine, I don't think we need to worry about the error since it still functions correctly :)

**Comment by schalkneethling** [comment]
@otterbotter you can now subscribe and unsubscribe from the settings page

**Comment by schalkneethling** [comment]
@otterbotter Also, you would need to use `yarn start` so that the service worker is in place.

**Comment by otterbotter** [comment]
> Understood, but I commented out my setting in my local env so, there should be no pushpad id either locally no? Do we always have a pushpad id set locally?

Ah, I ran into the same issue now but in reverse. Didn't have `PUSHPAD_PROJECT_ID` set so pushpad wasn't initialising. 

But, just populating the env var and re-running `yarn start` didn't work because the project id is injected during the webflow import in `src/js/webflow/import.js`

So to change the project id you actually need to either re-run the webflow import or just manually change it in `index.html` temporarily.

I've just opted to manually change it in `index.html` while reviewing the PR since I am not sure what import this PR is on.

**Review by otterbotter — APPROVED** [review]
Beautiful, good stuff :)

**Pull request merged**